### DISCUSSION TITLE
What's the behavior for failed job?

### DISCUSSION THREAD

**Discussion by lazywei** [immich-app/immich#3901] [Q&A]
Hi, when I checked the job stats on the admin page, it seems there are some failed tasks in some of the jobs.

My questions are:
- Do I need to worry about these failures?
- What are the expected behavior for these failures? Will Immich automatically retry them or do I need to manually re-trigger it?
- Is there any way to check the failures / errors msg for these failing tasks?

![image](https://github.com/immich-app/immich/assets/2840571/557eebb6-142b-4163-a5e9-753679a75f15)

Thanks

**Comment by PaulRitzkat** [comment]
I've got the same question, tried "missing", but my 155 failed stay:
![grafik](https://github.com/immich-app/immich/assets/2079167/fa8f3619-68e0-438e-9d23-8ff3a1c9f31a)

**Reply by jrasm91** [reply]
Failed jobs aren't ever deleted right now. They should go away if you restart redis though. In the future we hope to display them somewhere.

**Reply by PaulRitzkat** [reply]
Good to know! Restarting redis doesn't clear the failed badge though, but if I know that I don't have to respont to the failed jobs, I'll try to ignore it ;) 

Having a log and an indication why they failed would be nice for troubleshooting (in my case it was probably DNS, microservices couldn't download the geo info)

**Reply by jrasm91** [reply]
Yup, exactly. It might require recreating the container in that case. That's the only place it is persisted though.

**Reply by timdonovanuk** [reply]
Hallo! I think from your comment you're looking to enhance this at some point, but just a suggestion that from a UX perspective: telling users something failed, but providing no further details or the ability to drill down into the actual error, is a bit of an anti-pattern. Maybe something to consider for the roadmap. Thanks :)

**Reply by zuqini** [reply]
It's still a bit unclear to me how users should handle failures. Is pressing "MISSING" sufficient to retry the failed jobs?

**Comment by ppskj178** [comment]
I've been having the same problem since my NFS mounted fileserver went down.
I've searched and searched, but all I can find is to check the logs, and I can't find any detailed instructions.
I checked the compose logs on my docker and didn't see anything out of the ordinary.
I'm disappointed that there is still no guide for such a common problem.
